src/test/java/unit/GcpNotebookControlled.java

```diff
@@ -273,6 +273,10 @@ private static void assertNotebookState(String resourceName, String notebookStat
    */
   static void assertNotebookState(String resourceName, String notebookState, UUID workspaceId)
       throws JsonProcessingException {
+    Preconditions.checkState(
+        !notebookState.equals("PROVISIONING"),
+        "Don't assert that state is PROVISIONING. This leads to flaky tests, because by the time you assert PROVISIONING, state might already be ACTIVE.");
```

Comment:
I'm confused. Wouldn't it also be flaky to assert that it's not provisioning anymore?

Reply:
That's a good point. I deleted this and instead made `assertNotebookState` private. So tests should only call `pollDescribeForNotebookState`, which isn't flaky.